benches/language.rs

```diff
@@ -0,0 +1,259 @@
+                sinks::socket::SocketSinkConfig::make_basic_tcp_config(out_addr.to_string()),
+            );
+
+            config
+        }),
+        ("wasm", {
+            let mut config = config::Config::builder();
+            config.add_source(
+                "in",
+                sources::socket::SocketConfig::make_basic_tcp_config(in_addr),
+            );
+            config.add_transform(
+                "parser",
+                &["in"],
+                toml::from_str::<transforms::wasm::WasmConfig>(
+                    r#"
+module = "tests/data/wasm/vrl_announcement_example/target/wasm32-wasi/release/vrl_announcement_example.wasm"
+artifact_cache = "target/artifacts/"
+"#,
+                )
+                .unwrap(),
+            );
+            config.add_sink(
+                "out",
+                &["parser"],
+                sinks::socket::SocketSinkConfig::make_basic_tcp_config(out_addr.to_string()),
+            );
+
+            config
+        }),
+    ];
+
+    let mut group = c.benchmark_group("language/vrl_announcement");
+    group.sampling_mode(SamplingMode::Flat);
+
+    for (name, config) in configs.iter() {
+        group.throughput(Throughput::Elements(num_lines as u64));
+        group.bench_function(format!("language/vrl_announcement/{}", name), |b| {
+            b.iter_batched(
+                || {
+                    let config = config.clone();
+
+                    let mut rt = runtime();
+                    let (output_lines, topology) = rt.block_on(async move {
+                        let output_lines = CountReceiver::receive_lines(out_addr);
+                        let (topology, _crash) =
+                            start_topology(config.build().unwrap(), false).await;
+                        wait_for_tcp(in_addr).await;
+                        (output_lines, topology)
+                    });
+                    let lines = rt.block_on(async move {
+                        // TODO randomize lines
+                        //stream::repeat(vector::sources::util::fake::syslog_5424_log_line())
+                            //.take(num_lines)
+                            //.collect::<Vec<_>>()
+                            //.await
+                        stream::repeat(r#"<12>3 2020-12-19T21:48:09.004Z initech.io su 4015 ID81 - TPS report missing cover sheet"#.to_string()).take(num_lines).collect::<Vec<_>>().await
+                    });
+                    (rt, lines, topology, output_lines)
+                },
+                |(mut rt, lines, topology, output_lines)| {
+                    rt.block_on(async move {
+                        send_lines(in_addr, lines).await.unwrap();
+
+                        topology.stop().compat().await.unwrap();
+
+                        let output_lines = output_lines.await;
+
+                        #[cfg(debug_assertions)]
+                        {
+                            // TODO assert lines are tranformed
```

Comment:
Yeah. We should definitely make sure of this. Just in case one of the examples isn’t wired up correctly.